proposals/lip-Introduce_PoA_module.md

```diff
@@ -0,0 +1,607 @@
+* Each substore key is a 20-byte value `address`, where `address` is the address of the user account registered as validator, either in the genesis block or with an [authority registration command](#authority-registration-command).
+* Each substore value is the serialization of an object following the JSON schema `validatorObjectSchema` defined below.
+
+##### JSON Schema
+
+```java
+validatorObjectSchema = {
+   "type": "object",
+   "properties": {
+      "name": {
+         "dataType": "string",
+         "fieldNumber": 1
+      }
+   },
+   "required": [
+      "name"
+   ]
+}
+```
+
+##### Properties and Default Values
+
+`name` is a string representing the validator name, with a minimum length of 1 character and a maximum length of `MAX_LENGTH_NAME` characters. Its value is set in the genesis block or with an [authority registration command](#authority-registration-command)
+
+#### Name Substore
+
+The name substore is an auxiliary store used to validate the [authority registration command](#authority-registration-command).
+
+##### Store Prefix, Store Keys, and Store Values
+
+* The store prefix is set to `STORE_PREFIX_NAME`.
+* Each substore key is a name of a validator as given in the genesis block or with an [authority registration command](#authority-registration-command), serialized as a utf-8 encoded string.
+* Each substore value is set to the address of the corresponding validator, serialized according to the JSON schema `validatorAddressSchema` below.
+
+##### JSON Schema
+
+``` java
+validatorAddressSchema = {
+   "type": "object",
+   "properties": {
+      "address": {
+         "dataType": "bytes",
+         "fieldNumber": 1
+      }
+   },
+   "required": [
+      "address"
+   ]
+}
+```
+
+##### Properties and Default Values
+
+`address` is a 20-byte array with the address of the user account registered as validator in the genesis block or with an [authority registration command](#authority-registration-command).
+
+#### Snapshot Substore
+
+This substore contains the snapshot of the active authorities for the current round, next round and in two rounds.
+
+##### Store Prefix, Store Key, and Store Value
+
+* The store prefix is set to `STORE_PREFIX_SNAPSHOT`.
+* Each store key is `uint32be(roundNumber)`, where `roundNumber` can be 0, 1 or 2 corresponding to the current round, the next round and in two rounds respectively.
+* Each store value is the serialization of an object following `snapshotStoreSchema`.
+* Notation: Let `snapshotStore(roundNumber)` be the store entry with prefix STORE_PREFIX_SNAPSHOT and key `uint32be(roundNumber)`.
+
+##### JSON Schema
+
+```java
+snapshotStoreSchema = {
+   "type": "object",
+   "properties": {
+       "validators: {
+          "type": "array",
+          "fieldNumber": 1,    
+          "items": {
+              "type": "object",
+              "required": ["address", "weight"],
+              "properties": {
+	          "address": {
```

Comment:
I think here tabs and spaces are mixed and the indentation is incorrect in the source.

Reply:
fixed